app/models/manageiq/providers/redhat/infra_manager/provision_workflow.rb

```diff
@@ -92,4 +92,39 @@ def allowed_storages(options = {})
 
     result.select { |s| s.storage_domain_type == "data" }
   end
+
+  def load_allowed_vlans(hosts, vlans)
+    ems = source_ems
+    if update_vnic_profile_supported?(ems)
+      load_v4_allowed_vlans(ems, vlans)
+    else
+      super(hosts, vlans)
+    end
+  end
+
+  def source_ems
+    src = get_source_and_targets
+    load_ar_obj(src[:ems])
+  end
+
+  def update_vnic_profile_supported?(ems)
+    ems.supports_update_vnic_profile?
+  end
+
+  def load_v4_allowed_vlans(ems, vlans)
+    ems.with_provider_connection(:version => 4) do |connection|
```

Comment:
Also we don't usually go directly to the API here, but rather refresh the data and then show it from a MIQ DB. This code will run in UI worker, then the potential issues are, that it can be blocking the worker, or the appliance with UI worker is not configured for reaching the Provider.

@agrare do you think we can do this as a first version, before the refresh is done, or we tend not to do this, because of the issues I've mentioned and possibly others?